src/core/InstrumentSoundShaping.cpp

```diff
@@ -352,27 +346,72 @@ void InstrumentSoundShaping::loadSettings( const QDomElement & _this )
 	m_filterEnabledModel.loadSettings( _this, "fwet" );
 
 	QDomNode node = _this.firstChild();
-	while( !node.isNull() )
+	while (!node.isNull())
 	{
-		if( node.isElement() )
+		if (node.isElement())
 		{
-			for (auto i = std::size_t{0}; i < NumTargets; ++i)
+			const auto nodeName = node.nodeName();
+			if (nodeName == getVolumeNodeName())
 			{
-				if( node.nodeName() ==
-					m_envLfoParameters[i]->nodeName() +
-					QString( targetNames[i][1] ).
-								toLower() )
-				{
-					m_envLfoParameters[i]->restoreState( node.toElement() );
-				}
+				getVolumeParameters().restoreState(node.toElement());
+			}
+			else if (nodeName == getCutoffNodeName())
+			{
+				getCutoffParameters().restoreState(node.toElement());
+			}
+			else if (nodeName == getResonanceNodeName())
+			{
+				getResonanceParameters().restoreState(node.toElement());
 			}
 		}
+
 		node = node.nextSibling();
 	}
 }
 
+const EnvelopeAndLfoParameters& InstrumentSoundShaping::getVolumeParameters() const
+{
+	return m_volumeParameters;
+}
 
+EnvelopeAndLfoParameters& InstrumentSoundShaping::getVolumeParameters()
+{
+	return m_volumeParameters;
+}
 
+const EnvelopeAndLfoParameters& InstrumentSoundShaping::getCutoffParameters() const
+{
+	return m_cutoffParameters;
+}
+
+EnvelopeAndLfoParameters& InstrumentSoundShaping::getCutoffParameters()
+{
+	return m_cutoffParameters;
+}
 
+const EnvelopeAndLfoParameters& InstrumentSoundShaping::getResonanceParameters() const
+{
+	return m_resonanceParameters;
+}
+
+EnvelopeAndLfoParameters& InstrumentSoundShaping::getResonanceParameters()
+{
+	return m_resonanceParameters;
+}
```

Comment:
I think these definitions can go in the header, like for the other model getters.

Reply:
Done with commit https://github.com/LMMS/lmms/pull/7229/commits/09ac77ba756b53338cc5e6486affd6a46088728f.